.github/workflows/sync-opencraft-release-palm-1.yml

```diff
@@ -0,0 +1,50 @@
+name: Sync opencraft-release/palm.1 with upstream
+
+# Controls when the workflow will run
+on:
+# TODO: Uncomment this out when its ready to be merged
+#   schedule:
+#     - cron: '0 0 * * 1'  # Runs every Monday 00:00 UTC
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    env:
+      # TODO: set this to the OpenCraft Requirements Bot PAT
+      GITHUB_TOKEN: ${{ secrets.SYNC_GH_ACTIONS_PAT }}
+      # The fork repo provided needs to be in the {owner}/{repo} format
+      FORK_REPO: open-craft/edx-platform
+      UPSTREAM_REPO: https://github.com/tester-org-1/tester-repo
+      FORK_RELEASE: opencraft-release/palm.1
+      UPSTREAM_MASTER: master
+
+    steps:
+      - name: Get and store current datetime stamp
+        id: current-datetime-stamp
+        run: echo "DATE_TIME_STAMP=$(date +'%Y%m%d-%s')" >> "$GITHUB_OUTPUT"
+      - name: Configure git to use GitHub Cli for credentials
+        run: gh auth setup-git -h github.com
+      - name: Checkout Forked Repo
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.FORK_REPO }}
+          token: ${{ secrets.SYNC_GH_ACTIONS_PAT }}
+      - name: Add Remote Upstream
+        run: git remote add upstream $UPSTREAM_REPO
+      - name: Fetch Latest from Upstream
+        run: git fetch upstream
+      - name: Create a new branch off Upstream master
+        run: git checkout -b sync-$UPSTREAM_MASTER-${{ steps.current-datetime-stamp.outputs.DATE_TIME_STAMP }} upstream/$UPSTREAM_MASTER
+      - name: Push new branch to Fork
+        run: git push origin
```

Comment:
One more note - we should also check if there are any changes between branches. We don't want to open empty PRs.